mdantic/mdantic.py

```diff
@@ -74,38 +79,44 @@ def get_related_enum_helper(ty, visited, result):
                 get_related_enum_helper(sub_ty, visited, result)
 
 
-def mk_struct(cls, structs):
+def mk_struct(cls: BaseModel, structs: Dict[str, List[Field]]) -> None:
     this_struct = []
     structs[cls.__name__] = this_struct
-    for _, f in cls.__fields__.items():
-        ty = f.type_
-        description = f.field_info.description or ""
-        related_enums = get_related_enum(ty)
+    for field_name, f in cls.model_fields.items():
+        title = f.title or field_name
+        annotation = f.annotation
+        description = "" if f.description is None else f.description
+
+        related_enums = get_related_enum(annotation)
         if related_enums:
             for e in related_enums:
                 description += f"</br>{e.__name__}: {get_enum_values(e)}"
-        default = str(f.default if f.default is not None else "")
-        if hasattr(f, "_type_display"):
-            ty = f._type_display()
-        elif hasattr(ty, "__name__"):
-            ty = ty.__name__
+
+        default = f.get_default()  # str(f.default if f.default is not None else "")
+        default = None if str(default) == "PydanticUndefined" else str(default)
+
+        if hasattr(annotation, "__origin__"):
+            ty = str(annotation)
+        elif hasattr(annotation, "__name__"):
+            ty = annotation.__name__
```

Comment:
I've added a annotation is None check that returns None. We might want to print a warning when this happens, though I think its probably fairly uncommon since most of the point of something like pydantic involves adding type hints.